File: 5-matematica/1138.cpp
#include<iostream>
#include<cstring>
#include<cmath>
//
int store[10][10]={},digitsA[10],digitsB[10],lengthA,lengthB;
void splitNumber(int x,int receiver[],int&storeLength){
	storeLength=0;
	while(x){
		receiver[storeLength]=x%10;
		x/=10;
		++storeLength;
	}
}
int minusZeros(int lengthNumber){
	int result=0;
	for(int i=0;i<lengthNumber+1;++i)result+=std::pow(10,i);
	return result-1;
}
int regenerateNumber(int storeDigits[],int index){
	int result=0;
	for(int i=0;i<index+1;++i)result+=storeDigits[i]*std::pow(10,i);
	return result;
}
int*interval(int index,int tight,int storeDigits[],int digit){
	if(index<0){
		int*rest=(int*)calloc(10,sizeof(int));
		rest[digit]=1;
		return rest;
	}
	if(store[index][digit]&&!tight)return store[index];
	int rep=(tight)?storeDigits[index]:9,*amountDigits=(int*)calloc(10,sizeof(int));
	for(int i=0;i<rep+1;++i){
		int newTight=(i==storeDigits[index])?tight:0;
		int*temp=interval(index-1,newTight,storeDigits,i);
		for(int j=0;j<10;++j)amountDigits[j]+=temp[j];
		if(newTight&&index)amountDigits[i]+=regenerateNumber(storeDigits,index-1)+1;
		else if(index)amountDigits[i]+=std::pow(10,index);
	}
	for(int i=0;i<10&&!tight;++i)store[index][i]+=amountDigits[i];
	return amountDigits;
}
//
int main(){
	int a,b;
	memset(store,sizeof(store),0);
	while(std::cin>>a>>b&&(a|b)){	
		//
		lengthB=0;
		splitNumber(b,digitsB,lengthB);
		int*tempB=interval(lengthB-1,1,digitsB,0);
		(lengthB>1)?tempB[0]-=minusZeros(lengthB-1):tempB[0];
		splitNumber(a,digitsA,lengthA);
		//
		lengthA=0;
		splitNumber(a-1,digitsA,lengthA);
		int*tempA=interval(lengthA-1,1,digitsA,0);
		(lengthA>1)?tempA[0]-=minusZeros(lengthA-1):tempA[0];
		for(int i=0;i<9;++i)std::cout<<tempB[i]-tempA[i]<<" ";
		std::cout<<tempB[9]-tempA[9]<<"\n";
	}
}

File: 5-matematica/1138.py
storeDigits=None
def splitNum(x,storeNum)->None:
    while x:
        storeNum.append(x%10);
        x//=10
def regenerateNum(index,storeNum)->int:
    result=0
    for i in range(index+1):result+=storeNum[i]*10**i
    return result
def interval(tight,index,storeNum)->int:
    if index<0:
        ret=[0 for i in range(10)]
        return ret
    if storeDigits[index][9] and not tight:return storeDigits[index];
    rep=9 if not tight else storeNum[index]
    amountDigits=[0 for i in range(10)]
    for i in range(rep+1):
        #print('{})'.format(i),end='')
        newTight=tight if i==storeNum[index] else 0
        temp=interval(newTight,index-1,storeNum)
        for j in range(10):amountDigits[j]+=temp[j]
        amountDigits[i]+=10**(index)if not newTight else regenerateNum(index-1,storeNum)+1
    if not tight and not storeDigits[index][9]:
        for i in range(10):storeDigits[index][i]+=amountDigits[i]
    #print('{}'.format(str(amountDigits)[1:-1]),end='|')
    return amountDigits
def minusZeros(index)->int:
    result=0
    if index<2:return result
    for i in range(index):
        result+=10**i
    return result-1
#
storeDigits=[[0 for i in range(10)] for j in range(10)]
while 1:
    [a,b]=list(map(int,input().split()))
    if not a and not b:break
    digitsB=[]
    splitNum(b,digitsB)
    tempB=interval(1,len(digitsB)-1,digitsB)
    tempB[0]-=minusZeros(len(digitsB))
    #
    digitsA=[]
    splitNum(a-1,digitsA)
    if not(len(digitsA)):digitsA.append(0)
    tempA=interval(1,len(digitsA)-1,digitsA)
    tempA[0]-=minusZeros(len(digitsA))
    for i in range(9):print('{}'.format(tempB[i]-tempA[i]),end=' ')
    print(tempB[9]-tempA[9])
